src/main/java/org/opensearch/knn/index/util/Lucene.java

```diff
@@ -17,16 +17,43 @@
 import java.util.Map;
 import java.util.function.Function;
 
+import static org.opensearch.knn.common.KNNConstants.DYNAMIC_CONFIDENCE_INTERVAL;
+import static org.opensearch.knn.common.KNNConstants.ENCODER_SQ;
+import static org.opensearch.knn.common.KNNConstants.LUCENE_SQ_BITS;
+import static org.opensearch.knn.common.KNNConstants.LUCENE_SQ_CONFIDENCE_INTERVAL;
+import static org.opensearch.knn.common.KNNConstants.LUCENE_SQ_DEFAULT_BITS;
+import static org.opensearch.knn.common.KNNConstants.MAXIMUM_CONFIDENCE_INTERVAL;
+import static org.opensearch.knn.common.KNNConstants.METHOD_ENCODER_PARAMETER;
 import static org.opensearch.knn.common.KNNConstants.METHOD_HNSW;
 import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION;
 import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_M;
+import static org.opensearch.knn.common.KNNConstants.MINIMUM_CONFIDENCE_INTERVAL;
 
 /**
  * KNN Library for Lucene
  */
 public class Lucene extends JVMLibrary {
 
     Map<SpaceType, Function<Float, Float>> distanceTransform;
+    private static final List<Integer> LUCENE_SQ_BITS_SUPPORTED = List.of(7);
+
+    private final static Map<String, MethodComponent> HNSW_ENCODERS = ImmutableMap.of(
+        ENCODER_SQ,
+        MethodComponent.Builder.builder(ENCODER_SQ)
+            .addParameter(
+                LUCENE_SQ_CONFIDENCE_INTERVAL,
+                new Parameter.DoubleParameter(
+                    LUCENE_SQ_CONFIDENCE_INTERVAL,
+                    null,
```

Comment:
Why null default?

Reply:
For Lucene, the default value for confidence_interval is null which will later compute the confidence_interval based on dimension of vector (1-1/(1+d))